renku/core/management/session/docker.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018-2022 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Docker based interactive session provider."""
+
+from pathlib import Path
+from typing import Any, Dict, Iterable, List, Optional, Tuple
+from uuid import uuid4
+
+import docker
+
+from renku.core import errors
+from renku.core.management.client import LocalClient
+from renku.core.models.session import ISessionProvider, Session
+from renku.core.plugins import hookimpl
+
+
+class DockerSessionProvider(ISessionProvider):
+    """A docker based interactive session provider."""
+
+    JUPYTER_PORT = 8888
+
+    def __init__(self):
+        self._docker_client = None
+
+    def docker_client(self) -> docker.client.DockerClient:
```

Comment:
```suggestion
    @property
    def docker_client(self) -> docker.client.DockerClient:
```

Reply:
yeah i wanted this, but cannot recall why i had to resolve to use this... i'll change it locally and will report the whys